### DISCUSSION TITLE
Incorrect code generation due to naming edge case

### DISCUSSION THREAD

**Discussion by ckl-bso** [microsoft/powerplatform-build-tools#561] [General]
Hi,

we are using pac modelbuilder in version 1.27.6+g9f62afb of the pac cli to generate code.
In one of our projects the generated code does not compile due to inconsistent code. I assume this is due to a name conflict resolution scheme.

We have the following situation (as a reproduction, not the real names)

| Dataverse element| Schemaname | 
|--------|--------|
| Table| crb28_BusinessTypeGroup| 
| Column in Table (Type Choice, linked to Global Choice) | crb28_BusinessTypeGroupCode| 
| Global Choice | crb28_businesstypegroup| 

The code which is generated sums up to this (condensed to the relevant information)

```
namespace SDK
{
	[System.Runtime.Serialization.DataContractAttribute()]
	[Microsoft.Xrm.Sdk.Client.EntityLogicalNameAttribute("crb28_businesstypegroup")]
	[System.CodeDom.Compiler.GeneratedCodeAttribute("Dataverse Model Builder", "2.0.0.6")]
	public partial class crb28_BusinessTypeGroup : Microsoft.Xrm.Sdk.Entity, System.ComponentModel.INotifyPropertyChanging, System.ComponentModel.INotifyPropertyChanged
	{				
		[Microsoft.Xrm.Sdk.AttributeLogicalNameAttribute("crb28_businesstypegroupcode")]
		public virtual crb28_businesstypegroupEnum? crb28_BusinessTypeGroupCode
		{
			get
			{
				return ((crb28_businesstypegroupEnum?)(EntityOptionSetEnum.GetEnum(this, "crb28_businesstypegroupcode")));
			}
			set
			{
				this.OnPropertyChanging("crb28_BusinessTypeGroupCode");
				this.SetAttributeValue("crb28_businesstypegroupcode", value.HasValue ? new Microsoft.Xrm.Sdk.OptionSetValue((int)value) : null);
				this.OnPropertyChanged("crb28_BusinessTypeGroupCode");
			}
		}
	}

	
	[System.Runtime.Serialization.DataContractAttribute()]
	[System.CodeDom.Compiler.GeneratedCodeAttribute("Dataverse Model Builder", "2.0.0.6")]
	public enum crb28_businesstypegroup
	{
		
		[System.Runtime.Serialization.EnumMemberAttribute()]
		A = 198580000,
		
		[System.Runtime.Serialization.EnumMemberAttribute()]
		B = 198580001,
		
		[System.Runtime.Serialization.EnumMemberAttribute()]
		C = 198580002,
	}
}
```
This does not compile as there is no type `crb28_businesstypegroupEnum` defined, as the choice is represented as `crb28_businesstypegroup`.
I assume the postfix `Enum` is added in the type reference of the property to mitigate the name conflict with the class name.

I know that the naming of the elements is odd, but we have to live with that in this case. Nevertheless, I would expect that the generated code should be consistent in itself.

It is not really hindering us, however if you often update the generated code it gets a little tedious.
If it would be helpful I could provide you a solution with a repro including the used config for the modelbuilder.

**Comment by petrochuk** [comment]
It would be better to open issue here: [PowerPlatform-DataverseModelBuilder](https://github.com/microsoft/PowerPlatform-DataverseModelBuilder). PAC CLI is using this library.

**Comment by ckl-bso** [comment]
Thanks for the hint @petrochuk. I've just opened an issue there https://github.com/microsoft/PowerPlatform-DataverseModelBuilder/issues/6
